ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/StartContinuousHikingTransitionCondition.java

```diff
@@ -0,0 +1,26 @@
+package us.ihmc.behaviors.activeMapping.ContinuousHikingStateMachine;
+
+import behavior_msgs.msg.dds.ContinuousWalkingCommandMessage;
+import us.ihmc.behaviors.activeMapping.ContinuousHikingParameters;
+import us.ihmc.robotics.stateMachine.core.StateTransitionCondition;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public class StartContinuousHikingTransitionCondition implements StateTransitionCondition
+{
+   private final AtomicReference<ContinuousWalkingCommandMessage> commandMessage;
+   private final ContinuousHikingParameters continuousHikingParameters;
+
+   public StartContinuousHikingTransitionCondition(AtomicReference<ContinuousWalkingCommandMessage> commandMessage,
+                                                   ContinuousHikingParameters continuousHikingParameters)
+   {
+      this.commandMessage = commandMessage;
+      this.continuousHikingParameters = continuousHikingParameters;
+   }
+
+   @Override
+   public boolean testCondition(double timeInCurrentState)
+   {
+      return continuousHikingParameters.getEnableContinuousWalking() && commandMessage.get().getEnableContinuousWalking();
```

Comment:
Only way to start ContinuousHiking is if the checkbox is checked, and the CTRL key is being held